src/rebar_prv_upgrade.erl

```diff
@@ -82,6 +82,11 @@ do_(State) ->
     Deps = [Dep || Dep <- TopDeps ++ ProfileDeps, % TopDeps > ProfileDeps
                    is_atom(Dep) orelse is_atom(element(1, Dep))],
     Names = parse_names(rebar_utils:to_binary(proplists:get_value(package, Args, <<"">>)), Locks),
+
+    %% TODO: Should we check for updates here?
+    %% I think it makes sense to start doing so since we don't pull the whole registry anymore
+    %% [rebar_packages:update_package(Name, State) || Name <- Names],
```

Comment:
yeah agreed. Since we don't pull the whole registry, it might be cheap to do an update first, and then proceed as usual.